x/mm2/keeper/msg_server.go

```diff
@@ -142,6 +142,41 @@ func (ms msgServer) Params(goCtx context.Context, msg *types.MsgParams) (*types.
 	return &types.MsgParamsResponse{}, nil
 }
 
+func (ms msgServer) RemoveMarketAuthorities(goCtx context.Context, msg *types.MsgRemoveMarketAuthorities) (*types.MsgRemoveMarketAuthoritiesResponse, error) {
+	if msg == nil {
+		return nil, fmt.Errorf("unable to process nil msg")
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	params, err := ms.k.GetParams(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if msg.Admin != params.Admin {
+		return nil, fmt.Errorf("request admin %s does not match module admin %s", msg.Admin, params.Admin)
+	}
+
+	if len(msg.RemoveAddresses) > len(params.MarketAuthorities) {
+		return nil, fmt.Errorf("remove addresses must be a subset of the current market authorities")
```

Comment:
Not covered by tests.

Reply:
https://github.com/skip-mev/slinky/pull/293/commits/3754c0555ca61a38ac96d44d74bc48c5a37837d2